File: lab02/lab02/Result.cs
namespace lab02;

public class Result
{
    public int TotalValue { get; private set; }
    public int TotalWeight { get; private set; }
    public List<Item> Items { get; private set; }

    public Result(List<Item> items)
    {
        var totalValue = 0;
        var totalWeight = 0;
        foreach (var itm in items)
        {
            totalValue += itm.Value;
            totalWeight += itm.Weight;
        }

        TotalValue = totalValue;
        TotalWeight = totalWeight;
        Items = items;
    }

    public override string ToString()
    {
        var itemsString = string.Join(", ", Items);
        return $"Result(totalValue={TotalValue}, totalWeight={TotalWeight}, items=[{itemsString}])";
    }
}

File: lab06/Knapsack/src/main/java/com/raczu/knapsack/Result.java
package com.raczu.knapsack;

import java.util.List;

public class Result {
    public int totalValue;
    public int totalWeight;
    public final List<Item> items;

    public Result(List<Item> items) {
        this.totalValue = 0;
        this.totalWeight = 0;

        for (Item item : items) {
            this.totalValue += item.value;
            this.totalWeight += item.weight;
        }
        this.items = items;
    }

    @Override
    public String toString() {
        return String.join("\n", this.items.stream().map(Item::toString).toList());
    }
}
